### PR TITLE
Add FXIOS-9056 - Add configuration to toolbar button to render text inside image

### PR DESCRIPTION
**Pull request by PARAIPAN9**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9056)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/20050)

## :bulb: Description

- This PR adds the `TabNumberButton` class, responsible for displaying the number of open tabs. Currently in draft status because the animation has not yet been added.

| Normal  | Highlighted | 
| ------------- | ------------- |
|normal | highlighted|
## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by isabelrios** [comment]
Please use latest changes in main to have your PR running. Thanks!

File: BrowserKit/Sources/ToolbarKit/TabNumberButton.swift [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import Common
+
+final class TabNumberButton: ToolbarButton {
+    // MARK: - UX Constants
+    struct UX {
+        static let cornerRadius: CGFloat = 2
+        static let titleFont = UIFont.boldSystemFont(ofSize: 11)
```
Comment (thatswinnie) [inline BrowserKit/Sources/ToolbarKit/TabNumberButton.swift:12]: We should use `FXFontStyles` here.

File: BrowserKit/Sources/ToolbarKit/TabNumberButton.swift [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        // Tab count related constants
+        static let defaultCountLabelText = "0"
+        static let maxTabCountToShowInfinity = 100
```
Comment (thatswinnie) [inline BrowserKit/Sources/ToolbarKit/TabNumberButton.swift:16]: According to the logic later in code the last number displayed is 99. I suggest to change the constant here and adjust the logic accordingly.

File: BrowserKit/Sources/Common/Constants/StandardImageIdentifiers.swift [hunk 3]
```diff
@@ -101,6 +101,7 @@ public struct StandardImageIdentifiers {
         public static let subtract = "subtractLarge"
         public static let sync = "syncLarge"
         public static let syncTabs = "syncTabsLarge"
+        public static let tabNumber = "tabNumberLarge"
```
Comment (thatswinnie) [inline BrowserKit/Sources/Common/Constants/StandardImageIdentifiers.swift:104]: Where is this image coming from? The Acorn icon library doesn't have an image named this way. Only images found in Acorn should be added to StandardImageIdentifiers. Please use `tab` instead.

Comment (PARAIPAN9) [reply BrowserKit/Sources/Common/Constants/StandardImageIdentifiers.swift:104]: I hadn't found it, so I took it from the Client.

**Review by thatswinnie — APPROVED** [review]

**Pull request merged**